src/main.cairo

```diff
@@ -1,84 +1,140 @@
+use openzeppelin::{
+    token::erc721::{ERC721Component::{ERC721Metadata, HasComponent}},
+    introspection::src5::SRC5Component,
+};
+use custom_uri::{
+    main::custom_uri_component::InternalImpl as CustomURIInternalImpl, main::custom_uri_component
+};
+
+#[starknet::interface]
+trait IERC721Metadata<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn token_uri(self: @TState, tokenId: u256) -> Array<felt252>;
+    fn tokenURI(self: @TState, tokenId: u256) -> Array<felt252>;
+}
+
+#[starknet::embeddable]
+impl IERC721MetadataImpl<
+    TContractState,
+    +HasComponent<TContractState>,
+    +SRC5Component::HasComponent<TContractState>,
+    +custom_uri_component::HasComponent<TContractState>,
+    +Drop<TContractState>
+> of IERC721Metadata<TContractState> {
+    fn name(self: @TContractState) -> felt252 {
+        let component = HasComponent::get_component(self);
+        ERC721Metadata::name(component)
+    }
+
+    fn symbol(self: @TContractState) -> felt252 {
+        let component = HasComponent::get_component(self);
+        ERC721Metadata::symbol(component)
+    }
+
+    fn token_uri(self: @TContractState, tokenId: u256) -> Array<felt252> {
+        let component = custom_uri_component::HasComponent::get_component(self);
+        let hundred: NonZero<u256> = 100_u256.try_into().unwrap();
+        let big_number: NonZero<u256> = 0x2000000_u256.try_into().unwrap();
+        let (quotient, remainder) = DivRem::div_rem(tokenId, hundred);
+        if (remainder != 99) {
+            return component.get_uri(remainder);
+        } else {
+            let (value, digit) = DivRem::div_rem(quotient, big_number);
+            return component.get_uri(digit);
+        }
+    }
+
+    fn tokenURI(self: @TContractState, tokenId: u256) -> Array<felt252> {
```

Comment:
Please call token_uri, that way we avoid duplicate code and thus the risk of changing bug only in half of the places